docs/en/agent/performance-tests.md

```diff
@@ -1,3 +1,94 @@
 # Performance Tests
 
-Currently, this part has not been done yet.
\ No newline at end of file
+Performance testing is used to verify the impact on application performance when using SkyWalking Go.
+
+## Test Objective
+
+By launching both the **agent** and **non-agent** compiled applications, we subject them to the same **QPS** under stress testing, 
+evaluating the CPU, memory, and network latency of the machine during the testing period.
+
+The application has been saved and submitted to the [test/performance](../../../test/performance) directory, with the following topology:
+
+```
+pressure -> consumer -> provider
+```
+
+The stress testing service (pressure) uses multithreading to send HTTP requests to the consumer service. 
+When the consumer receives a request, it sends three requests to the provider service to obtain return data results. 
+Based on these network requests, when using SkyWalking Go, the consumer service generates four Spans (1 Entry Span, 3 Exit Spans).
+
+### Application
+
+The application's integration with SkyWalking Go follows the same process as other applications. For more information, please [refer to the documentation](../setup/gobuild.md).
+
+In the application, we use loops and mathematical calculations (`math.Log`) to simulate the execution of the business program. 
+This consumes a certain amount of CPU usage, preventing idle processing during service stress testing and amplifying the impact of the Agent program on the business application.
+
+### Stress Testing Service
+
+We use the [Vegeta](https://github.com/tsenart/vegeta) service for stress testing, which launches traffic at a specified QPS to the application. 
+It is based on the Go language and uses goroutines to provide a more efficient stress testing solution.
+
+## Test Environment
+
+A total of 4 GCP machines are launched, all instances are 4C8G.
+
+1. **pressure**: Used for deploying traffic to the consumer machine.
+2. **consumer**: Used for deploying the consumer service.
+3. **provider**: Used for deploying the provider service.
+4. **skywalking**: Used for deploying the SkyWalking backend cluster, providing a standalone OAP node (in-memory H2 storage) and a UI interface.
+
+Each service is deployed on a separate machine to ensure there is no interference with one another.
+
+## Test Process
+
+### Preparation Phase
+The preparation phase is used to ensure that all machines and test case preparations are completed.
+
+#### Pressure
+Install the [Vegeta service](https://github.com/tsenart/vegeta#install) on the stress testing instance and create the following file(`request.txt`) to simulate traffic usage.
+
+```
+GET http://${CONSUMER_IP}:8080/consumer
+Sw8: 1-MWYyZDRiZjQ3YmY3MTFlYWI3OTRhY2RlNDgwMDExMjI=-MWU3YzIwNGE3YmY3MTFlYWI4NThhY2RlNDgwMDExMjI=-0-c2VydmljZQ==-aW5zdGFuY2U=-cHJvcGFnYXRpb24=-cHJvcGFnYXRpb246NTU2Ng=="
```

Comment:
Is the header changed or not?

Reply:
The header does not change.

Reply:
The segment is request thread oriented, right?
So, generally, although all segments are the same trace, it is still good for agent side.

Reply:
Yes. Although all segments belong to the same trace, the agent must process all headers in all requests.

Reply:
OK, got it. Actually, the Java agent doesn't process this part in benchmark.